File: fix_layout.py
import os
import re

files = [
    'src/pages/conferences/ICCSD.tsx',
    'src/pages/conferences/GSAI.tsx',
    'src/pages/conferences/WCCC.tsx',
    'src/pages/conferences/ICBT.tsx',
    'src/pages/conferences/IDSS.tsx'
]

for file in files:
    with open(file, 'r', encoding='utf-8') as f:
        content = f.read()

    # Step 1: Find Registration Sidebar
    reg_start = content.find('{/* Right Column: Registration Sidebar */}')
    
    # We want to grab everything from reg_start down to the end of the sidebar wrapper div.
    # The sidebar wrapper ends right before the closing of the grid:
    #             </div>
    #           </div>
    #         </div>
    #       </section>
    
    reg_regex = re.compile(r'\{\/\* Right Column: Registration Sidebar \*\/\}\s*<div.*?relative">\s*<motion\.div.*?>.*?<\/p>\s*<\/div>\s*<\/motion\.div>\s*<\/div>', re.DOTALL)
    reg_match = reg_regex.search(content)
    
    if not reg_match:
        print("Could not find Registration Sidebar in", file)
        continue
        
    registration_code = reg_match.group(0)
    
    # Remove registration code from its original position
    new_content = content.replace(registration_code, '')
    
    # We also need to fix the closing tags. Wait, let's just do a string replacement around the About / Tracks border.
    
    # Let's find:
    #                 </div>
    #               </motion.div>
    # 
    #               {/* Tracks/Themes Cards */}
    
    about_end = re.compile(r'(<\/div>\s*<\/motion\.div>)\s*(\{\/\* Tracks\/Themes Cards \*\/\})', re.DOTALL)
    new_content = about_end.sub(r'\1\n            </div>\n\n            ' + registration_code.replace('\\', '\\\\') + r'\n          </div>\n\n          <div className="space-y-24 mt-24">\n            \2', new_content, 1)

    # Finally, remove the extra closing tags left behind at the bottom where Registration used to live.
    # Original end was:
    #               </motion.div>
    # 
    #             </div>
    #             
    #           </div>
    #         </div>
    #       </section>
    
    # It will now look like:
    #               </motion.div>
    # 
    #             </div>
    #             
    #             
    #           </div>
    #         </div>
    #       </section>
    # We should delete one `</div>` wrapper because we closed `grid lg:grid-cols-3` already.

    cleanup_regex = re.compile(r'(<\/div>\s*<\/motion\.div>)\s*<\/div>\s*<\/div>\s*<\/div>\s*<\/section>', re.DOTALL)
    new_content = cleanup_regex.sub(r'\1\n          </div>\n        </div>\n      </section>', new_content)

    with open(file, 'w', encoding='utf-8') as f:
        f.write(new_content)
        
    print("Fixed", file)

File: fix_layout.js
const fs = require('fs');
const path = require('path');

const dir = path.join(__dirname, 'src', 'pages', 'conferences');
const files = ['ICCSD.tsx', 'GSAI.tsx', 'WCCC.tsx', 'ICBT.tsx', 'IDSS.tsx'];

for (const file of files) {
  const filePath = path.join(dir, file);
  let content = fs.readFileSync(filePath, 'utf-8');

  // Step 1: Find where the Right Column starts.
  const rightColumnMarker = '{/* Right Column: Registration Sidebar */}';
  const regexRightColumn = new RegExp(`\\s*\\/\\* Right Column: Registration Sidebar \\*\\/[\\s\\S]*?By registering, you agree to our Terms of Service \\& Privacy Policy\\.\\n\\s*<\\/p>\\n\\s*<\\/div>\\n\\s*<\\/motion\\.div>\\n\\s*<\\/div>`);
  
  const rightMatch = content.match(regexRightColumn);
  if (!rightMatch) {
    console.log(`Could not find Right Column in ${file}`);
    continue;
  }
  
  const rightColumnCode = rightMatch[0];
  
  // Step 2: Remove the right column from its original place.
  let newContent = content.replace(rightColumnCode, '');

  // We also need to remove the closing div of the grid which comes right after the rightColumnCode!
  // It looks like:
  //          </div>
  //        </div>
  //      </section>
  // Actually, wait: the right column was followed by:
  //             </div>
  //             
  //           </div>
  //         </div>
  //       </section>
  
  // Let's refine the replacement strategy to be safer.
  
  const regexAboutEnd = new RegExp(`(\\/\\* About Section \\*\\/[\\s\\S]*?<\\/motion\\.div>)`);
  const aboutMatch = newContent.match(regexAboutEnd);
  
  if (aboutMatch) {
    const aboutSectionCode = aboutMatch[0];
    
    // Instead of doing complicated splits, let's just assemble it manually.
    // The structure:
    // ...
    //           <div className="grid lg:grid-cols-3 gap-12 lg:gap-16">
    //             <div className="lg:col-span-2 space-y-24">
    //               {/* About Section */} ... </motion.div>
    // ^ INSERT RIGHT COLUMN HERE, close the  <div className="grid... ">, open <div className="space-y-24 mt-24">
    //               {/* Tracks/Themes Cards */} ...
    //               {/* Why Attend Section */} ...
    //               {/* Speakers Section */} ... </motion.div>
    //             </div> (closing lg:col-span-2)
    // ^ REMOVE THIS div, because we already closed the grid and we are in our own full-width div now.
    
    const splitPoint = newContent.indexOf('              {/* Tracks/Themes Cards */}');
    
    const beforeSplit = newContent.substring(0, splitPoint);
    const afterSplit = newContent.substring(splitPoint);
    
    // In beforeSplit, there is:
    // <div className="lg:col-span-2 space-y-24">
    // ...about...
    // </motion.div>
    // <split point>
    
    // So right before split, we close the lg:col-span-2 div, insert the right column, close the grid div, then start a new wrapper div.
    
    let injected = beforeSplit 
       + `            </div>\n` 
       + `            ${rightColumnCode}\n`
       + `          </div>\n\n`
       + `          <div className="space-y-24 mt-24">\n`;
       
    // Now for afterSplit, it will end with:
    //               </div>
    //             </div>
    //             
    //           </div>
    //         </div>
    //       </section>
    
    // We inserted the right column and closed the grid.
    // So the trailing code in afterSplit needs one less `</div>`.
    
    // Let's just find the `            </div>\n            \n          </div>` and remove one level.
    let finalContent = injected + afterSplit;
    
    // Remove the trailing grid wrap which is now empty.
    // Around the end:
    //               </motion.div>
    //             </div> (this is the closing of lg:col-span-2 in original)
    //             
    //           </div> (this is the closing of grid in original)
    //         </div> (this is container-custom)
    
    const cleanupRegex = /<\/motion\.div>\s*\n\s*<\/div>\s*\n\s*<\/div>\s*\n\s*<\/div>\s*\n\s*<\/section>/;
    const cleanMatch = finalContent.match(cleanupRegex);
    if(cleanMatch){
         finalContent = finalContent.replace(cleanupRegex, `</motion.div>\n            </div>\n        </div>\n      </section>`);
         fs.writeFileSync(filePath, finalContent);
         console.log(`Updated ${file} successfully.`);
    } else {
         console.log(`Cleanup match failed in ${file}`);
    }

  } else {
    console.log(`Could not find About Section in ${file}`);
  }
}
